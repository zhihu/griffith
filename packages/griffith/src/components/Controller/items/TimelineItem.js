import React from 'react'
import {css} from 'aphrodite/no-important'
import styles from '../styles'
import Timeline from '../../Timeline'

const TimelineItem = props => {
  return (
    <div className={css(styles.timeline)}>
      <Timeline {...props} />
    </div>
  )
}

export default TimelineItem
