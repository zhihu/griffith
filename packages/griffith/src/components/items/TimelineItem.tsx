import React from 'react'
import {css} from 'aphrodite/no-important'
import styles from '../Controller.styles'
import Timeline, {TimelineProps} from '../Timeline'

const TimelineItem = (props: TimelineProps) => {
  return (
    <div className={css(styles.timeline)}>
      <Timeline {...props} />
    </div>
  )
}

export default TimelineItem
