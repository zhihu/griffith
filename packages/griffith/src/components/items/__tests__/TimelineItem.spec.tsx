import React from 'react'
import {render} from '@testing-library/react'
import TimelineItem from '../TimelineItem'

describe('TimelineItem', () => {
  it('get TimelineItem component', () => {
    const result = render(
      <TimelineItem
        value={7}
        total={187}
        buffered={12}
        onDragStart={jest.fn()}
        onDragEnd={jest.fn()}
        onChange={jest.fn()}
        onSeek={jest.fn()}
      />
    )
    expect(result.container).toMatchSnapshot()
  })
})
