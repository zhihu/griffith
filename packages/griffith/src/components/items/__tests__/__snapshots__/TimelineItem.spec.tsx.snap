// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem get TimelineItem component 1`] = `
<div>
  <div class="timeline_bc4egv">
    <div class="root_3hmsj">
      <div
        class="root_17bkmwy-o_O-root_1u52a97-o_O-root_1pr59fm"
        tabindex="0"
        role="slider"
      >
        <div class="inner_9gm5i8-o_O-inner_qb00m9">
          <div class="track_1bopucd-o_O-track_2wh41e-o_O-track_1q3h7g7">
            <div
              class="
                bar_1kpj1ed-o_O-bar_g2nzkb-o_O-bar_1kvrquw-o_O-buffered_gsgvhl
              "
              style="left: 0px; width: 6.417112299465241%"
            ></div>
            <div
              class="bar_1kpj1ed-o_O-bar_g2nzkb-o_O-bar_1kvrquw"
              style="left: 0px; width: 3.7433155080213902%"
            ></div>
          </div>
          <div
            class="thumbWrapper_ay4wjb-o_O-thumbWrapper_6t4rjm"
            style="left: 3.7433155080213902%"
          >
            <div class="thumb_ozdqky-o_O-thumb_1omjhz8"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>

`;
